shared/bls/bls.go

```diff
@@ -221,3 +259,25 @@ func Domain(domainType []byte, forkVersion []byte) uint64 {
 	b = append(b, forkVersion[:4]...)
 	return bytesutil.FromBytes8(b)
 }
+
+// HashWithDomain hashes 32 byte message and uint64 domain parameters a Fp2 element
+func HashWithDomain(messageHash [32]byte, domain [8]byte) []byte {
+	xReBytes := [41]byte{}
+	xImBytes := [41]byte{}
+	xBytes := make([]byte, 96)
+	copy(xReBytes[:32], messageHash[:])
+	copy(xReBytes[32:40], domain[:])
+	copy(xReBytes[40:41], []byte{0x01})
+	copy(xImBytes[:32], messageHash[:])
+	copy(xImBytes[32:40], domain[:])
+	copy(xImBytes[40:41], []byte{0x02})
+	copy(xBytes[16:48], hash(xImBytes[:]))
+	copy(xBytes[64:], hash(xReBytes[:]))
+	return xBytes
+}
+
+func hash(in []byte) []byte {
```

Comment:
Please delete this, use https://github.com/prysmaticlabs/prysm/blob/master/shared/hashutil/hash.go#L18